floodlight/io/statsperform.py

```diff
@@ -427,3 +432,683 @@ def read_open_statsperform_tracking_data_csv(
+    away_ht2 = XY(xy=xydata["Away"][2], framerate=framerate)
+    ball_ht1 = XY(xy=xydata["Ball"][1], framerate=framerate)
+    ball_ht2 = XY(xy=xydata["Ball"][2], framerate=framerate)
+
+    data_objects = (home_ht1, home_ht2, away_ht1, away_ht2, ball_ht1, ball_ht2)
+
+    return data_objects
+
+
+def read_event_data_from_url(
+    url_events: str,
+) -> Tuple[Events, Events, Events, Events, Pitch]:
+    """Reads a URL from the StatsPerform API (StatsEdgeViewer) containing an events csv
+    file and extracts the stored event data and pitch information.
+
+    Parameters
+    ----------
+    url_events: str
+        URL to the xml file containing the event data.
+
+    Returns
+    -------
+    data_objects: Tuple[Events, Events, Events, Events, Pitch]
+        Events-objects for both teams and both halves and pitch information. The order
+        is (home_ht1, home_ht2, away_ht1, away_ht2, pitch).
+    """
+    data_dir = os.path.join(DATA_DIR, "statsperform")
+    temp_file = "events_temp.xml"
+    download_event_data_from_url(url_events=url_events, filename_xml=temp_file)
+    data_objects = read_event_data_xml(filepath_xml=os.path.join(data_dir, temp_file))
+    os.remove(os.path.join(data_dir, temp_file))
+    return data_objects
+
+
+def read_tracking_data_from_url(
+    url_tracking: str,
+    links: Dict[str, Dict[int, int]] = None,
+) -> Tuple[XY, XY, XY, XY, XY, XY]:
+    """Reads a URL from the StatsPerform API (StatsEdgeViewer) containing a tracking
+    data txt file and extracts position data.
+
+    The tracking from the URL is downloaded into a temporary file stored in the
+    repository's internal root ``.data``-folder that is removed afterwards.
+
+    Parameters
+    ----------
+    url_tracking: str or pathlib.Path
+        URL to the txt file containing the tracking data.
+    links: Dict[str, Dict[int, int]], optional
+        A link dictionary of the form ``links[team][jID] = xID``. Player's are
+        identified in StatsPerform files via jID, and this dictionary is used to map
+        them to a specific xID in the respective XY objects. Should be supplied if that
+        order matters. If None is given (default), the links are automatically extracted
+        from the CSV file at the cost of a second pass through the entire file.
+
+    Returns
+    -------
+    data_objects: Tuple[XY, XY, XY, XY, XY, XY]
+        XY- and Pitch-objects for both teams and both halves. The order is
+        (home_ht1, home_ht2, away_ht1, away_ht2, ball_ht1, ball_ht2, pitch)
+    """
+    data_dir = os.path.join(DATA_DIR, "statsperform")
+    temp_file = "tracking_temp.txt"
+    download_tracking_data_from_url(url_tracking=url_tracking, filename_txt=temp_file)
+    data_objects = read_tracking_data_txt(
+        filepath_txt=os.path.join(data_dir, temp_file),
+        links=links,
+    )
+    os.remove(os.path.join(data_dir, temp_file))
+    return data_objects
+
+
+def download_event_data_from_url(
```

Comment:
first and foremost: this is not a statsperform-specific function and ideally should therefore be of general applicability and located in the ``io.utils`` submodule. unless there's a very good reason it only belongs here? I'll comment below as if it was a general function

Reply:
In this sense, it would also be necessary to identify the differences to the ``download_from_url`` function. What's the necessity for an iterative reading of lines? Which assumptions on the downloaded files are attached to this?

Reply:
I see your point about the last four functions not really belonging in the statsperform module. 

Although I still think that it could be useful to have a function that downloads a statsperform URL into the .data/statsperform folder without needing a manual interference (file movement etc.). But maybe this is rather a general functionality like

datasets.download_provider_file(url_file=..., provider_name="statsperform") -> creates .data/statsperform/filename_as_stored_in_url

Reply:
Regarding the iterative read this circumvents an error with XML files: 

"lxml.etree.XMLSyntaxError: Start tag expected, '<' not found, line 1, column 1"